example_gh_test.go

```diff
@@ -172,16 +211,58 @@ func ExampleGQLClient_Mutate_simple() {
 			}
 		} `graphql:"addStar(input: $input)"`
 	}
+	type AddStarInput struct {
+		StarrableID string `json:"starrableId"`
+	}
 	variables := map[string]interface{}{
-		"input": githubv4.AddStarInput{
```

Comment:
I'm a bit torn: on one hand I do appreciate that the new example has fewer dependencies, but I also did find value in this example demonstrating the use of the shurcooL library that ensures typed inputs for each GraphQL mutation. I would actually recommend the use of that library to API clients such as extensions. @mntlty: thoughts? https://github.com/cli/go-gh/pull/99

Reply:
I think there's two use cases of what the examples here are for. 
- To show how `go-gh` can be used
- To show how `go-gh` should be used

If it's the first, having another package when a `struct` will do the job isn't necessary. If it's the second, having `githubv4` here introduces it to users who will likely benefit from the structure and safety it provides. They don't have to use it, but it's a big help to have the compile time checks.

Personally I learned about `githubv4` when contributing to the CLI, which not everyone is fortunate enough to be able to do ❤️

Reply:
My thoughts on this are that as `go-gh` is meant to be used in other projects and all of our decencies come with it so we should try hard to minimize them. If `go` had a way to mark a dependency for test or examples only then this wouldn't matter. This was a case I felt that having the dependency was not worth it. What I think would be a good compromise would be to add a comment to this example where the struct is defined saying that the user could alternatively use the `githubv4` package where these are already defined. What do people think about that idea?

Reply:
@samcoe I like that compromise!